common/components/ui/Modal.jsx

```diff
@@ -0,0 +1,107 @@
+// @flow
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import closeIcon from 'assets/images/icon-x.svg';
+
+import './Modal.scss';
+
+export default class Modal extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool,
+    size: PropTypes.oneOf(['small', 'medium', 'large', 'xlarge']),
```

Comment:
I'm not sure how I would do that. If children components define an explicit width, that would work. But that won't work for responsive content, as doing something like `max-width: 600px; width: 100%;` won't correctly set the width. Example:

<img width="577" alt="screen shot 2017-07-14 at 4 04 40 pm" src="https://user-images.githubusercontent.com/649992/28228844-3222286a-68ae-11e7-9ecd-ea88cbd30674.png">